zcash_client_sqlite/src/wallet.rs

```diff
@@ -461,30 +515,78 @@ pub fn rewind_to_height<P: consensus::Parameters>(
                     .or(Ok(sapling_activation_height - 1))
             })?;
 
-    // nothing to do if we're deleting back down to the max height
-    if block_height >= last_scanned_height {
-        Ok(())
+    let rewind_height = if block_height >= (last_scanned_height - PRUNING_HEIGHT) {
+        Some(block_height)
     } else {
```

Comment:
This could be simplified to
```suggestion
    if block_height < last_scanned_height - PRUNING_HEIGHT {
```
and then unconditionally set `rewind_height = block_height` if we haven't returned.